### PR TITLE
Create Deck Page 

### PR DESCRIPTION
**Pull request by hidesminimally**
### Goals 
- Create Deck Routes
- Create Deck page that fetches Deck objects from DB 
Demo showing 
1. uploading anki deck (screen capture doesn't capture it) 
2. Seeing settings
3. Refreshing to see the new decks on nav 
screenrecording

### Future Improvements, Suggestions, and Alternatives 
Next: Render Deck Information and cards related to the deck 
@lbkchen Are you interested in helping w/ rendering the cards on this page? 

App requiring refresh for decks to show in nav is not good. We should probably think about React Context and/or MobX or something to fix that. Not sure of any other quick fixes. @lbkchen any ideas?

### CODE REVIEW

File: src/pages/my-app/decks/[deck].tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import Head from 'next/head';
+import { AdaptedAnkiDeck } from '../../../db/models';
+import db from '../../../db/db';
+
+const DynamicDeck: FunctionComponent = () => {
+  // This logic is copied whenever we need a deck. Should we use another mechanism (react context?)
+  const [ankiDeck, setAnkiDeck] = useState<AdaptedAnkiDeck>(null);
+  useEffect(() => {
```
Comment (hidesminimally) [inline src/pages/my-app/decks/[deck].tsx:12]: Realized a little late into writing the logic for rendering cards for a given deck, that we need to either (1) query cards from both our own cards DB and Anki DB or (2) make sure that the AnkiDB's cards schema conforms to our engram card DB. 

Option 1 is only a short-cut and we'll eventually have to think of a way to make the anki DB cards conform to our schema. In addition to the piling on of tech debt, we'll have to deal with the problems of having two types floating around constantly. 

The same problem also extends to decks. The logic here is incomplete. Not only do we have to query from Anki decks, but we also have to query from Engram decks. 

Better to solve the problem of creating an Engram schema earlier on. I'm putting this PR on hold to figure that out @lbkchen but wanted to give you an update.

Comment (hidesminimally) [reply src/pages/my-app/decks/[deck].tsx]: Done. this PR is ready to be reviewed

File: src/db/db.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ class DB {
   private sql: SqlJsStatic;
 
   constructor() {
+    debugger;
```
Comment (lbkchen) [inline src/db/db.ts]: should we remove this? this will always breakpoint here when the inspector is open

File: src/pages/my-app/decks/[deck].tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import Head from 'next/head';
+import { AdaptedAnkiDeck } from '../../../db/models';
+import db from '../../../db/db';
+
+const DynamicDeck: FunctionComponent = () => {
+  // This logic is copied whenever we need a deck. Should we use another mechanism (react context?)
+  const [ankiDeck, setAnkiDeck] = useState<AdaptedAnkiDeck>(null);
```
Comment (lbkchen) [inline src/pages/my-app/decks/[deck].tsx]: we can use a custom hook to encapsulate and reuse any logic within hooks!

**Review by lbkchen — APPROVED** [review]

**Comment by lbkchen** [comment]
> Next: Render Deck Information and cards related to the deck
> @lbkchen Are you interested in helping w/ rendering the cards on this page?

Yup down

> App requiring refresh for decks to show in nav is not good. We should probably think about React Context and/or MobX or something to fix that. Not sure of any other quick fixes. @lbkchen any ideas?

yeah react context is designed to solve problems like this where you want the state to be shared across several components in different areas of the component tree. We can even use simple `EventEmitter` implementations for more performance-critical use cases (react context will trigger rerenders in the whole trees where context is added, but with `EventEmitter` you can highly isolate the components that should receive certain events). I can take a look into it this week

File: src/pages/my-app/decks/[deck].tsx (hunk 3 above)
Comment (hidesminimally) [reply src/pages/my-app/decks/[deck].tsx]: will look into it, leaving a TODO for now. thanks for the suggestion

**Pull request merged**